flow/connectors/clickhouse/normalize.go

```diff
@@ -266,19 +267,84 @@ func (c *ClickHouseConnector) NormalizeRecords(
 		return nil, fmt.Errorf("failed to copy avro stages to destination: %w", err)
 	}
 
+	rawTbl := c.getRawTableName(req.FlowJobName)
+	for batchID := normBatchID + 1; batchID <= req.SyncBatchID; batchID++ {
+		err = c.syncTablesInThisBatch(ctx, req, rawTbl, batchID)
+		if err != nil {
+			c.logger.Error("[clickhouse] error while syncing tables in this batch", slog.Any("error", err),
+				slog.Int64("batchID", batchID))
+			return nil, err
+		}
+
+		err = c.UpdateNormalizeBatchID(ctx, req.FlowJobName, batchID)
+		if err != nil {
+			c.logger.Error("[clickhouse] error while updating normalize batch id",
+				slog.Any("error", err),
+				slog.Int64("batchID", batchID))
+			return nil, err
+		}
+	}
+
+	return &model.NormalizeResponse{
+		Done:         true,
+		StartBatchID: normBatchID + 1,
+		EndBatchID:   req.SyncBatchID,
+	}, nil
+}
+
+func (c *ClickHouseConnector) getDistinctTableNamesInBatch(
+	ctx context.Context,
+	flowJobName string,
+	batchID int64,
+) ([]string, error) {
+	rawTbl := c.getRawTableName(flowJobName)
+
+	q := fmt.Sprintf(
+		`SELECT DISTINCT _peerdb_destination_table_name FROM %s WHERE _peerdb_batch_id = %d`,
+		rawTbl, batchID)
+
+	rows, err := c.query(ctx, q)
+	if err != nil {
+		return nil, fmt.Errorf("error while querying raw table for distinct table names in batch: %w", err)
+	}
+	defer rows.Close()
+	var tableNames []string
+	for rows.Next() {
```

Comment:
Use pgx.CollectRows

Reply:
this is a query on ClickHouse connector, don't think that's possible